### ISSUE TITLE
useless_let_if_seq should not warn when variable is set from multiple places.

### ISSUE THREAD

**Issue by kevincox** [rust-lang/rust-clippy#3769]
I have some code that looks like this.

```rust
let ok = true;

if check1() {
  // Log some info.
  ok = false;
}

if check2() {
  // Log some info.
  ok = false;
}

if check3() {
  // Log some info.
  ok = false;
}
```

However clippy gives me a `useless_let_if_seq` saying that I should merge the first if into the `let`. I don't think this is a helpful suggestion because 1) It makes the first check look special and 2) it distracts a bit from the `true` until marked `false`.

After thinking about this a bit I think that if the variable defined from the `let` is set from multiple locations in the code then this warning should probably be suppressed.

**Comment by flip1995** [comment]
Duplicate of #3768 and #2749

**Comment by kevincox** [comment]
These are different. The linked bug is about setting multiple variables where this bug is about setting one variable from multiple locations.

**Comment by flip1995** [comment]
Oh yeah, you're right! Reopening it, sorry.
